R/Spectra.R

```diff
@@ -1028,6 +1028,8 @@ setMethod("filterPrecursorScan", "Spectra",
 #' @rdname Spectra
 setMethod("filterRt", "Spectra",
           function(object, rt = numeric(), msLevel. = unique(msLevel(object))) {
+              if (!identical(length(rt), 2L) || rt[1] >= rt[2])
```

Comment:
I know this is a small function that won't be called often but isn't `length(rt) != 2L` enough? Also this should check for `is.numeric`. Otherwise the user could use `characters` (won't raise an error) or `Date`s.
```suggestion
              if (length(rt) != 2L || !is.numeric(rt) || rt[1] >= rt[2])
```

Reply:
You are right. I wrongly assumed `rt` was defined as a numeric in the signature. But
```r
> filterRt(sps, c(1, 2))
MSn data (Spectra) with 0 spectra in a MsBackendDataFrame backend:
Processing:
 Filter: select retention time [1..2] on MS level(s)  [Tue Jun 18 20:46:26 2019] 
> filterRt(sps, c("1", "2"))
Error in .local(object, ...) : 'rt' must be a numeric
> traceback()
7: stop("'rt' must be a numeric")
6: .local(object, ...)
5: filterRt(object@backend, rt, msLevel.)
4: filterRt(object@backend, rt, msLevel.) at Spectra.R!7751MA3#6
3: .local(object, ...)
2: filterRt(sps, c("1", "2"))
1: filterRt(sps, c("1", "2"))
> getMethod("filterRt", "MsBackendDataFrame")
Method Definition:

function (object, ...) 
{
    .local <- function (object, rt = numeric(), msLevel = integer()) 
    {
        if (length(rt)) {
            rt <- range(rt)
            if (!is.numeric(rt)) 
                stop("'rt' must be a numeric")
            if (!length(msLevel)) 
                msLevel <- unique(msLevel(object))
            sel_ms <- msLevel(object) %in% msLevel
            sel_rt <- rtime(object) >= rt[1] & rtime(object) <= 
                rt[2] & sel_ms
            object[sel_rt | !sel_ms]
        }
        else object
    }
    .local(object, ...)
}
<bytecode: 0x55aa2b8cc168>
<environment: namespace:Spectra>

Signatures:
        object              
target  "MsBackendDataFrame"
defined "MsBackendDataFrame"
```
This is tested in the `MsBackendDataFrame` method. It should probably be defined once in `filterRt,Spectra`, shouldn't it?

This raises the question whether we would want to do these checks more systematically.

Reply:
Well, yes. Would make sense to do all input parameter tests in `Spectra`. As of now they are in the backends because I first implemented the backends and then `Spectra`, but we could move all tests up.

Reply:
Yes, let's aim for that. It makes much more sense, as the API is defined at the `Spectra` level anyway.